src/utils/FlowUtils.ts

```diff
@@ -69,18 +69,25 @@ export function filledSameDeposit(fillA: Fill, fillB: Fill): boolean {
 
 // Ensure that each deposit element is included with the same value in the fill. This includes all elements defined
 // by the depositor as well as the realizedLpFeePct and the destinationToken, which are pulled from other clients.
-export function validateFillForDeposit(fill: Fill, deposit?: Deposit, fillFieldsToIgnore: string[] = []): boolean {
+export function validateFillForDeposit(
+  fill: Fill,
+  deposit?: Deposit,
+  fillFieldsToIgnore: string[] = []
+): { valid: true } | { valid: false; reason: string } {
   if (deposit === undefined) {
-    return false;
+    return { valid: false, reason: "Deposit is undefined" };
   }
 
   // Note: this short circuits when a key is found where the comparison doesn't match.
   // TODO: if we turn on "strict" in the tsconfig, the elements of FILL_DEPOSIT_COMPARISON_KEYS will be automatically
   // validated against the fields in Fill and Deposit, generating an error if there is a discrepency.
-  return FILL_DEPOSIT_COMPARISON_KEYS.every((key) => {
-    if (fillFieldsToIgnore.includes(key)) {
-      return true;
+  for (const key of FILL_DEPOSIT_COMPARISON_KEYS) {
+    if (fill[key]?.toString() === deposit[key]?.toString() || fillFieldsToIgnore.includes(key)) {
```

Comment:
Minor nit: I wonder if the resulting code block would be more intuitive using a `find()` for the error case because that's what we're doing here.

Reply:
I migrated to using `find` here: https://github.com/across-protocol/sdk-v2/pull/493/commits/0820373148e5e70c036bcc348d9a9afe631c107b

I'd initially avoided it because it required inverting the negating all the comparison logic, but I do agree that it's an improvement :+1: